components/script/dom/servoparser/mod.rs

```diff
@@ -140,6 +144,72 @@ impl ServoParser {
         self.script_nesting_level.get()
     }
 
+    pub fn resume_with_pending_parsing_blocking_script(&self, script: &HTMLScriptElement) {
+        assert!(self.suspended.get());
+        self.suspended.set(false);
+
+        // This prepends the script input to the network input.
+        mem::swap(&mut *self.script_input.borrow_mut(), &mut *self.network_input.borrow_mut());
+        while let Some(chunk) = self.script_input.borrow_mut().pop_front() {
+            self.network_input.borrow_mut().push_back(chunk);
+        }
+
+        let script_nesting_level = self.script_nesting_level.get();
+        assert_eq!(script_nesting_level, 0);
+
+        self.script_nesting_level.set(script_nesting_level + 1);
+        script.execute();
+        self.script_nesting_level.set(script_nesting_level);
+
+        if !self.suspended.get() {
+            self.parse_sync();
+        }
+    }
+
+    pub fn write(&self, text: Vec<DOMString>) {
```

Comment:
Can we have a pointer to the part of the spec this is implementing? Is this step 8 of https://html.spec.whatwg.org/multipage/webappapis.html#dom-document-write?